services/src/user/user-service.ts

```
@@ -13,3 +15,28 @@ function convertUserToNonAuthUser(user: SelectableUser): User {
 
 	return nonAuthUser as User
 }
+
+export async function deleteUser(id: number): Promise<boolean> {
+	const result = await deleteUserById(id)
+
+	// TODO: add business logic to handle projects with or without other members
+
+	return result.numDeletedRows === 1n
+}
+
+export async function changeUserPassword(
+	id: number,
+	changePasswordPayload: ChangePasswordPayload
+): Promise<boolean> {
+	const user = await getUserById(id)
+
+	const passwordMatches = compare(changePasswordPayload.currentPassword, user.password_hash)
+	if (!passwordMatches) {
+		throw new Error('Invalid password')
+	}
+
+	const newPasswordHash = hash(changePasswordPayload.newPassword)
+	const updateResult = await changeUserPasswordById(id, newPasswordHash)
+
+	return updateResult.numUpdatedRows === 1n
```

Shouldn't this just throw an error if something goes wrong?
Like is there ever a case where this will return false?
Because if false indicates that something unexpected happened then it should throw an error